kratos/python_scripts/eigen_solver_factory.py

```diff
@@ -2,30 +2,33 @@
 
 def ConstructSolver(settings):
     import KratosMultiphysics
-    
+
     if(type(settings) != KratosMultiphysics.Parameters):
         raise Exception("Input is expected to be provided as a Kratos Parameters object")
-    
+
     solver_type = settings["solver_type"].GetString()
-    
-    if solver_type == "eigen_eigensystem":
-        import KratosMultiphysics.EigenSolversApplication
-        eigen_solver = KratosMultiphysics.EigenSolversApplication.EigensystemSolver(settings)
-        return eigen_solver
-    
-    import new_linear_solver_factory
-    linear_solver = new_linear_solver_factory.ConstructSolver(settings["linear_solver_settings"])
-        
+
+    if settings.Has("linear_solver_settings"):
+        import new_linear_solver_factory
+        linear_solver = new_linear_solver_factory.ConstructSolver(settings["linear_solver_settings"])
```

Comment:
In case the user forgets the ```linear_solver_settings```, to linear solver variable will not exist when it is passed to the constructor of the solvers.

Reply:
The version from before, was giving a proper error message about the missing parameter

Reply:
good point, I haven't thought abt this!

I will revert this or close this completely depending on the answer of riccardo and pooyan